src/Razor/test/Microsoft.VisualStudio.LanguageServerClient.Razor.Test/HtmlCSharp/FindAllReferencesHandlerTest.cs

```diff
@@ -609,7 +618,26 @@ public async Task HandleRequestAsync_LargeProject_InvokesCSharpLanguageServer()
 
             // Assert
             Assert.True(lspFarEndpointCalled);
-            Assert.Equal(NUM_BATCHES, progressBatchesReported);
+
+            var sortedBatchesReported = progressBatchesReported.ToList();
+            sortedBatchesReported.Sort((VSReferenceItem[] a, VSReferenceItem[] b) =>
```

Comment:
This sorting is possible due to the way we [created the batches in the first place](https://github.com/dotnet/aspnetcore-tooling/pull/2152/files#diff-a4cf52e0c9b8b9f834e6fd4068d04ebeL520) (sequentially numbered the range start character). This is just a testing detail rather than an implementation detail.